src/main/kotlin/com/hj/leetcode/kotlin/problem662/Solution.kt

```diff
@@ -0,0 +1,66 @@
+package com.hj.leetcode.kotlin.problem662
+
+import com.hj.leetcode.kotlin.common.model.TreeNode
+
+/**
+ * LeetCode page: [662. Maximum Width of Binary Tree](https://leetcode.com/problems/maximum-width-of-binary-tree/);
+ */
+class Solution {
+    /* Complexity:
+     * Time O(N) and Space O(N) where N is the number of nodes in root;
+     */
+    fun widthOfBinaryTree(root: TreeNode?): Int {
+        if (root == null) return 0
+
+        var maxWidth = 0L
+        /* In order to avoid number overflow(for example, in the case of a vertical zigzag tree), the indices
+         * of nodes are computed as following: During a BFS, when computing the indices of nodes in next level,
+         * the indices of nodes in current level are normalized such that the leftmost node in current level
+         * has an index of 0.
+         */
+        val nodeIndex = hashMapOf<TreeNode, Long>()
```

Comment:
The comment is unclear as it assumes that the reader is aware that we are computing the level-ordered index.

Reply:
resolved at ecd3feffaf050481559486c76c11613818dbad39